src/libqof/qof/gnc-datetime.hpp

```diff
@@ -153,10 +153,40 @@ class GncDateTime
     std::unique_ptr<GncDateTimeImpl> m_impl;
 };
 
+class GncDateFormat
+{
+public:
+    GncDateFormat (const char* fmt, const char* re) :
+    m_fmt(fmt), m_re(re) {}
+    const std::string m_fmt;
+private:
+    const std::string m_re;
+
+    friend class GncDateImpl;
+};
+
+/** Functor to check if a GncDateFormat refers to a specific date format.
+ * Intended to query GncDate::formats with std::find_if.
+ */
+struct GncDateFormatPred
```

Comment:
Yay, I just wrote my first lambda function :)
If we continue at this pace I'll be a c++ guru by the end of the month :D 
Unfortunately it still aborts during cleanup.